src/main/resources/lib/nextjs/connection-config.es6

```diff
@@ -28,18 +28,41 @@ export function getSiteConfig(pathOrId, repoId) {
     }
 }
 
-exports.getFrontendServerUrl = (config) => {
-    // read site config first
-    let url = config?.nextjsUrl;
+exports.getFrontendServerUrl = (config, project, site) => {
```

Comment:
A bit confused to why you added `project` and `site` here since you already have methods in the very same class for getting project (`getProjectName`), and site can be retrieved directly via portal lib. Also, these two arguments are only used if `app.config` is present, so they shouldn't be required.

Should be something like 

```Typescript
exports.getFrontendServerUrl = (config) => {
  let url;
  if (app.config) {
    const project = this.getProjectName();
    const siteName = portalLib.getSite()?._name || '';
    url = app.config[`settings.${project}.${siteName}.url`] || app.config['nextjs.url'];
  } else {
    url = config?.nextjsUrl;
  }

  return url?.replace(removeEndSlashPattern, '') || "http://localhost:3000";
```

Same with `getFrontendServerToken`

Reply:
Yeah, that's because I invoke `getProjectName` and `getSite` only once in the very beginning of the flow and then pass results in `requestContext` instead of invoking them every time they are needed.